src/olympia/versions/tests.py

```diff
@@ -1294,6 +1295,26 @@ def test_track_upload_time(self):
             assert (actual_delta >= (rough_delta - fuzz) and
                     actual_delta <= (rough_delta + fuzz))
 
+    def test_new_version_is_10s_compatible_no_feature_compat_previously(self):
+        assert not self.addon.feature_compatibility.pk
+        self.upload = self.get_upload('multiprocess_compatible_extension.xpi')
+        version = Version.from_upload(self.upload, self.addon,
+                                      [self.platform])
+        assert version.pk
+        assert self.addon.feature_compatibility.pk
+        assert self.addon.feature_compatibility.e10s == amo.E10S_COMPATIBLE
+
+    def test_new_version_is_10s_compatible(self):
+        AddonFeatureCompatibility.objects.create(addon=self.addon)
+        assert self.addon.feature_compatibility.e10s == amo.E10S_UNKNOWN
+        self.upload = self.get_upload('multiprocess_compatible_extension.xpi')
+        version = Version.from_upload(self.upload, self.addon,
+                                      [self.platform])
+        assert version.pk
+        assert self.addon.feature_compatibility.pk
+        self.addon.feature_compatibility.reload()
+        assert self.addon.feature_compatibility.e10s == amo.E10S_COMPATIBLE
+
```

Comment:
Maybe add another webextension test for the sake of testing all status? I know it's probably a bit redundant but you could potentially add e10s compat by releasing a webextension for an old add-on too afair.